src/Trakx.Kaiko.ApiClient.Tests/Integration/ReferenceDataTests.cs

```diff
@@ -51,4 +51,50 @@ public async Task InstrumentsClient_should_return_list_of_instruments()
         response.Result.Data.Should().NotBeNull();
         response.Result.Data.Should().HaveCountGreaterThan(0);
     }
+
+    [InlineData(false)]
+    [Theory]
+    public async Task Download_legacy_symbols(bool download)
+    {
+        if (!download)
+        {
+            download.Should().BeFalse();
+            return;
+        }
+
+        var client = ServiceProvider.GetRequiredService<IInstrumentsClient>();
+        var response = await client.GetAllInstrumentsAsync();
+
+        using var writer = new StreamWriter("kaiko-legacy-symbols.csv");
+
+        void Write(params string[] items)
+        {
+            var line = string.Join(',', items); // .Select(p => $"'{p}")
+            writer.WriteLine(line);
+        }
+
+        Write("kaiko_legacy_symbol", "base_asset", "quote_asset");
+
+        var seen = new HashSet<string>();
```

Comment:
It this `HashSet` being used? It is just declared and then is used to check if has some symbol in line 92. However, It seems to be always empty no?

Reply:
Thanks for catching. The DistinctBy() does the job the 'seen' HashSet was meant for. Removed now.